### ISSUE TITLE
support build stamp with docker_push

### ISSUE THREAD

**Issue by mikedanese** [bazelbuild/rules_docker#3]
I should be able to use a variable from my workspace status to tag a docker image when running docker_push.

**Comment by mattmoor** [comment]
`docker_push` supports make-style expansion for the components:

```
vardef("MY_TAG", "latest")

docker_push(
    name = "foo_push",
    image = ":foo_image",
    registry = "gcr.io",
    repository = "foo/bar/baz",
    tag = "$(MY_TAG)",
)
```

Or is that not what you want?

**Comment by mattmoor** [comment]
@damienmg How can I access [this](https://github.com/kubernetes/kubernetes/blob/master/.bazelrc#L6) in skylark?

**Comment by mattmoor** [comment]
Ok, blocked by: https://github.com/bazelbuild/bazel/issues/1054

**Comment by ixdy** [comment]
Similar / clarifying request: I'd like to be able to tag an image automatically using the git sha of the workspace. We use `--workspace_status_command` to set this for go rules, but there's currently no way to use that in docker rules.

Make-style expansion isn't ideal, since that requires adding `--define`s to the bazel command line.

**Comment by mattmoor** [comment]
In other places we allow `@foo` to load the value from a file.  I wonder if this would be sufficient for your purposes?

**Comment by ixdy** [comment]
That might work. I implemented a hack (https://github.com/ixdy/bazel/commit/3b29803eb528ff525c7024190ffbf4b08c598cf2) to do that a few months ago, and it's what the kubernetes bazel build is currently using, though it still feels pretty hacky

(I got preoccupied with other things and didn't follow up on some review feedback, and now things have moved around a bit so I'd need to rebase my change.)

The go rules got around waiting for Skylark support (https://github.com/bazelbuild/bazel/issues/1054) by implementing stamping through reading the `ctx.info_file` and `ctx.version_file` directly.

Since IIRC most of the actual docker image/layer-building implementation happens in python, I wonder if we could do something similar and have the info/version files passed through to the scripts on `stamp=1`? 
I haven't thought through all of the implications yet, or if the rules will work correctly with unexpanded variables in skylark.
